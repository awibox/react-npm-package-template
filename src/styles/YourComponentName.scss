$fontSize: 14px;
$color: #000;

.text {
  color: $color;
  font-size: $fontSize;
}
