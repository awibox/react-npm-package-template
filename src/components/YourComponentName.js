import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from '../styles/YourComponentName.scss';

class YourComponentName extends Component {
  static propTypes = {
    text: PropTypes.string,
  };

  static defaultProps = {
    text: 'Hello world',
  };

  render() {
    return (
      <div className={styles.text}>
        {this.props.text}
      </div>
    );
  }
}

export default YourComponentName;
