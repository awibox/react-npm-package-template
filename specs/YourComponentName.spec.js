import React from 'react';
import { shallow } from 'enzyme';
import YourComponentName from '../src/components/YourComponentName';

describe('YourComponentName', () => {
  it('should render correctly', () => {
    const component = shallow(<YourComponentName />);
    expect(component).toMatchSnapshot();
  });
});
