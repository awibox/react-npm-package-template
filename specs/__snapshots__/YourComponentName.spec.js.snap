// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YourComponentName should render correctly 1`] = `
<div
  className="text"
>
  Hello world
</div>
`;
